.github/workflows/pull_request.yml

```diff
@@ -14,13 +14,38 @@ on:
       - main
 
 jobs:
+  build-template:
+    uses: ./.github/workflows/build_test_template.yml
+    secrets:
+      E2B_TESTS_ACCESS_TOKEN: ${{ secrets.E2B_TESTS_ACCESS_TOKEN }}
+    with:
+      E2B_DOMAIN: ${{ vars.E2B_DOMAIN }}
   js-sdk:
     uses: ./.github/workflows/js_tests.yml
+    needs: build-template
+    if: always() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled')
```

Comment:
does the if need to be here? This is used in other workflows just because there possibility to skip the step, same for python